CMakeLists.txt

```diff
@@ -381,6 +381,20 @@ install_helper(
   FILES ${MANPAGES}
   DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
 
+# MIN_LOG_LEVEL for log.h
+if(DEFINED MIN_LOG_LEVEL)
+  if(NOT MIN_LOG_LEVEL MATCHES "^[0-3]$")
+    message(FATAL_ERROR "MIN_LOG_LEVEL must be a number DEBUG (0), INFO (1), WARNING (2) or ERROR (3)")
+  endif()
+  message(STATUS "Log level set to ${MIN_LOG_LEVEL}")
+elseif(TRAVIS_CI_BUILD)
```

Comment:
Why move this from `.travis.yml`? I think we should avoid sprinkling CI-specific conditions in the build files--try to keep it in .travis.yml.